File: Languages/npeg_javascript/TokenMatch.js

if(!RobustHaven) var RobustHaven={};
if(!RobustHaven.Text) RobustHaven.Text={};
if(!RobustHaven.Text.Npeg) RobustHaven.Text.Npeg={};

RobustHaven.Text.Npeg.TokenMatch = function (name, start, end) {
    var name = name,
        start = start,
        end = end;

    this.getName = function () {return name;};
    this.setName = function ( value ) {	name = value;	};

    this.getStart = function ()	{	return start;	};
    this.setStart = function ( value ) {	start = value;	};

    this.getEnd = function ()	{ return end; };
    this.setEnd = function ( value ){	end = value;	};
}

File: Languages/npeg_csharp/RobustHaven.Text.Npeg/TokenMatch.cs
using System;

namespace RobustHaven.Text.Npeg
{
	public class TokenMatch
	{
		public TokenMatch(String name, Int32 start, Int32 end, Byte[] value)
		{
			Name = name;
			Start = start;
			End = end;
			Value = value;
		}

		public String Name { private set; get; }
		public Int32 Start { private set; get; }

		public Int32 End { private set; get; }

		public Byte[] Value { get; set; }
	}
}